### ISSUE TITLE
GenericMemberIterator fallback different than forward declaration

### ISSUE THREAD

**Issue by englercj** [Tencent/rapidjson#1531]
If I define `RAPIDJSON_NOMEMBERITERATORCLASS` then `GenericMemberIterator` is declared as a struct but the forward delcaration still uses `class` so a warning is emitted.

```
1>...\include\rapidjson\fwd.h(108): error C2220: warning treated as error - no 'object' file generated
1>...\include\rapidjson\fwd.h(108): warning C4099: 'rapidjson::GenericMemberIterator': type name first seen using 'struct' now seen using 'class'
```

Either the forward declaration needs to switch with this define, or the definition needs to use `class` rather than `struct`.

**Comment by miloyip** [comment]
Fix in d87b698d0fcc10a5f632ecbc80a9cb2a8fa094a5

**Comment by englercj** [comment]
Wonderful, thanks!
